Sources/Authentication/Bearer/BearerAuthenticatable.swift

```diff
@@ -2,29 +2,26 @@ import Async
 import Bits
 import Fluent
 
-/// Authenticatable by `Bearer token` auth.
-public protocol BearerAuthenticatable: Authenticatable {
-    /// Key path to the token
-    typealias TokenKey = WritableKeyPath<Self, String>
-
-    /// The key under which the model's unique token is stored.
-    static var tokenKey: TokenKey { get }
-
-    /// Authenticates using the supplied credentials and connection.
-    static func authenticate(using bearer: BearerAuthorization, on connection: DatabaseConnectable) -> Future<Self?>
-}
+extension BearerAuthorization: HeaderValueAuthorization {}
 
-extension BearerAuthenticatable where Self: Model {
-    /// See `BearerAuthenticatable`.
-    public static func authenticate(using bearer: BearerAuthorization, on conn: DatabaseConnectable) -> Future<Self?> {
-        return Self.query(on: conn).filter(tokenKey == bearer.token).first()
-    }
+/// Authenticatable by `Bearer token` auth.
+public protocol BearerAuthenticatable: HeaderValueAuthenticatable where AuthorizationType == BearerAuthorization {
 }
 
-extension BearerAuthenticatable {
+// This extension captures both things explicitly declared to be
+// BearerAuthenticatable and also things declared to be
+// HeaderValueAuthenticatable with an AuthorizationType of
+// BearerAuthorization.
+extension HeaderValueAuthenticatable where AuthorizationType == BearerAuthorization {
```

Comment:
I think this would make more sense as an extension on `BearerAuthenticatable`

Reply:
Yeah, the more I think about it the more whole-heartedly I agree.